variants/aws-ecs-1/Cargo.toml

```diff
@@ -18,6 +18,7 @@ included-packages = [
 # core
     "release",
     "kernel-5.10",
+    "amazon-ssm-agent-plugin",
```

Comment:
You need to remove the `ecs-agent.spec` dependency on `amazon-ssm-agent`, or else you will automatically install that package on all ECS variants.

You should replace that dependency with the plugin dependency, so that `ecs-agent` takes care of its own requirements and variants don't have to specify them.